File: Olympiad/POI/official/2013/code/kon/konzaw/pas/kon.pas
unit kon;

interface

    procedure inicjuj(n, k : LongInt; var D : array of LongInt);

    procedure podlej(a, b : LongInt);

    function dojrzale(a, b : LongInt) : LongInt;

implementation

    var
        N, K : LongInt;
        t : array[0..300000] of LongInt;

    procedure inicjuj(n, k : LongInt; var D : array of LongInt);
    var
        i : LongInt;
    begin
        N := n; K := k;
        for i := 0 to n-1 do
            t[i] := D[i];
    end;

    procedure podlej(a, b : LongInt);
    begin
        inc(t[a]);
        inc(t[b]);
    end;

    function dojrzale(a, b : LongInt) : LongInt;
    begin
        dojrzale := 0;
        if (t[a] >= K) then inc(dojrzale);
        if (t[b] >= K) then inc(dojrzale);
        { cos glupiego }
    end;

begin
end.

File: Olympiad/POI/official/2013/code/kon/konzaw/cpp/kon.cpp
int N, K, t[10];

void inicjuj(int n, int k, int *D)
{
    N = n, K = k;
    for (int i = 0; i < n; ++i)
        t[i] = D[i];
}

void podlej(int a, int b)
{
    t[a] += 1;
    t[b] += 1;
}

int dojrzale(int a, int b)
{
    return (t[a] >= K) + (t[b] >= K); /* cos glupiego */
}
